target/linux/qualcommax/ipq807x/base-files/etc/board.d/02_network

```diff
@@ -55,9 +54,32 @@ ipq807x_setup_interfaces()
 	esac
 }
 
+ipq807x_setup_macs()
+{
+	local board="$1"
+	local lan_mac=""
+	local wan_mac=""
+	local label_mac=""
+
+	case "$board" in
+		linksys,mx5300)
+			label_mac=$(mtd_get_mac_ascii devinfo hw_mac_addr)
+			for i in $(seq 1 4); do
+				[ "$(mtd_get_mac_ascii u_env eth${i}addr)" != "$label_mac" ] && lan_mac=$label_mac
+			done
+			[ "$(mtd_get_mac_ascii u_env eth5addr)" != "$label_mac" ] && wan_mac=$label_mac
+		;;
+	esac
+
+	[ -n "$lan_mac" ] && ucidef_set_interface_macaddr "lan" $lan_mac
+	[ -n "$wan_mac" ] && ucidef_set_interface_macaddr "wan" $wan_mac
+	[ -n "$label_mac" ] && ucidef_set_label_macaddr $label_mac
```

Comment:
Where should the label MAC address be set so that it can be used in hotplug.d scripts?
If we do it here, the `get_mac_label` function does not return the correct address in hotplug.d scripts during firstboot.

Reply:
It can be set via 02_network or via an alias in DTS, depending on board

Reply:
Setting via DTS is not possible here.
The setting from `02_network` works correctly, but with `firstboot` the function `get_mac_label` called in the `hotplug.d` scripts returns `00:00:0*`.

What is the startup order?
1. init.d
2. hotplug.d
3. board.d

Reply:
The hotplug script probably starts before 02_board